### PR TITLE
cmake: binutils objdump disassemble all sections

### PR DESCRIPTION
**Pull request by rgujju**
Added support for creating an assembly listing containing all sections like rodata, data and debug sections, not just those expected to contain instructions. This will be helpful when there is important content in the data sections. Added a new config option OUTPUT_DISASSEMBLE_ALL to select between this and the original one. This new config option does will be 'n' by default.

### CODE REVIEW

File: cmake/bintools/gnu/target_objdump.cmake [hunk 1]
```diff
@@ -19,7 +19,7 @@ function(bintools_objdump)
     # Prefix of output variables
     BINTOOLS_OBJDUMP
     # List of argument names without values, hence boolean
-    "DISASSEMBLE;DISASSEMBLE_SOURCE"
+    "DISASSEMBLE;DISASSEMBLE_SOURCE;DISASSEMBLE_ALL"
```
Comment (tejlmand) [inline cmake/bintools/gnu/target_objdump.cmake:22]: You forgot to add `DISASSEMBLE_ALL` to description of the function `bintools_objdump` above ^^^

File: CMakeLists.txt [hunk 2]
```diff
@@ -1213,13 +1213,25 @@ endif()
...
+  else()
+    bintools_objdump(
+      RESULT_CMD_LIST    out_disassembly_cmd
+      RESULT_BYPROD_LIST out_disassembly_byprod
+      DISASSEMBLE_SOURCE
+      FILE_INPUT         ${KERNEL_ELF_NAME}
+      FILE_OUTPUT        ${KERNEL_LST_NAME}
+      )
+  endif()
+
```
Comment (tejlmand) [inline CMakeLists.txt:1234]: No need to supplicate the call to `bintools_objdump`
```suggestion
  set(disassemble_type DISASSEMBLE_SOURCE)
  if(CONFIG_OUTPUT_DISASSEMBLE_ALL)
    set(disassemble_type DISASSEMBLE_ALL)
  endif()
    
  bintools_objdump(
    RESULT_CMD_LIST    out_disassembly_cmd
    RESULT_BYPROD_LIST out_disassembly_byprod
    ${disassemble_type}
    FILE_INPUT         ${KERNEL_ELF_NAME}
    FILE_OUTPUT        ${KERNEL_LST_NAME}
    )
```

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: cmake/bintools/gnu/target_objdump.cmake (hunk 1 above)
Comment (rgujju) [reply cmake/bintools/gnu/target_objdump.cmake:23]: Done!

File: CMakeLists.txt (hunk 2 above)
Comment (rgujju) [reply CMakeLists.txt]: Done!

**Review by tejlmand — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**